### PR TITLE
Update ddp example

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Our ddp example in readme was missing import statements + some extra code to make it able to run.
Need confirmation from someone with more ddp experience that this is correct.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -123,48 +123,64 @@ Module metric usage remains the same when using multiple GPUs or multiple nodes.
...
+import torch
+from torch import nn
+import torch.distributed as dist
+import torch.multiprocessing as mp
+from torch.nn.parallel import DistributedDataParallel as DDP
+import torchmetrics
 
-# create default process group
-dist.init_process_group("gloo", rank=rank, world_size=world_size)
+def metric_ddp(rank, world_size)
```
Comment (awaelchli) [inline README.md:134]:
```suggestion
def metric_ddp(rank, world_size):
```

File: README.md [hunk 2]
```diff
@@ -123,48 +123,64 @@ Module metric usage remains the same when using multiple GPUs or multiple nodes.
...
+        print(f"Accuracy on all data: {acc}, accelerator rank: {rank}")
+
+        # Reseting internal state such that metric ready for new data
+        metric.reset()
+    
+    # cleanup
+    dist.destroy_process_group()
+
+world_size = 2   # number of gpus to parallize over
+mp.spawn(metric_dpp, args=(world_size,), nprocs=world_size, join=True)
```
Comment (awaelchli) [inline README.md:182]:
```suggestion
if __name__ == "__main__":
    world_size = 2   # number of gpus to parallize over
    mp.spawn(metric_ddp, args=(world_size,), nprocs=world_size, join=True)
```

**Review by awaelchli — COMMENTED** [review]
This didn't run for me because of syntax error and a missing guard.
See my suggestions. This is the output of the script:

```
Accuracy on batch 0: 0.20000000298023224
Accuracy on batch 1: 0.20000000298023224
Accuracy on batch 2: 0.20000000298023224
Accuracy on batch 3: 0.0
Accuracy on batch 4: 0.4000000059604645
Accuracy on batch 5: 0.20000000298023224
Accuracy on batch 6: 0.30000001192092896
Accuracy on batch 7: 0.0
Accuracy on batch 8: 0.10000000149011612
Accuracy on batch 9: 0.0
Accuracy on all data: 0.13500000536441803, accelerator rank: 0
Accuracy on all data: 0.13500000536441803, accelerator rank: 1
Accuracy on batch 0: 0.30000001192092896
Accuracy on batch 1: 0.20000000298023224
Accuracy on batch 2: 0.10000000149011612
Accuracy on batch 3: 0.10000000149011612
Accuracy on batch 4: 0.10000000149011612
Accuracy on batch 5: 0.0
Accuracy on batch 6: 0.20000000298023224
Accuracy on batch 7: 0.10000000149011612
Accuracy on batch 8: 0.20000000298023224
Accuracy on batch 9: 0.20000000298023224
Accuracy on all data: 0.17000000178813934, accelerator rank: 0
Accuracy on all data: 0.17000000178813934, accelerator rank: 1
Accuracy on batch 0: 0.30000001192092896
Accuracy on batch 1: 0.20000000298023224
Accuracy on batch 2: 0.10000000149011612
Accuracy on batch 3: 0.10000000149011612
Accuracy on batch 4: 0.20000000298023224
Accuracy on batch 5: 0.0
Accuracy on batch 6: 0.20000000298023224
Accuracy on batch 7: 0.20000000298023224
Accuracy on batch 8: 0.30000001192092896
Accuracy on batch 9: 0.20000000298023224
Accuracy on all data: 0.1899999976158142, accelerator rank: 0
Accuracy on all data: 0.1899999976158142, accelerator rank: 1
Accuracy on batch 0: 0.30000001192092896
Accuracy on batch 1: 0.0
Accuracy on batch 2: 0.0
Accuracy on batch 3: 0.10000000149011612
Accuracy on batch 4: 0.20000000298023224
Accuracy on batch 5: 0.30000001192092896
Accuracy on batch 6: 0.4000000059604645
Accuracy on batch 7: 0.10000000149011612
Accuracy on batch 8: 0.10000000149011612
Accuracy on batch 9: 0.30000001192092896
Accuracy on all data: 0.1550000011920929, accelerator rank: 0
Accuracy on all data: 0.1550000011920929, accelerator rank: 1
Accuracy on batch 0: 0.20000000298023224
Accuracy on batch 1: 0.0
Accuracy on batch 2: 0.0
Accuracy on batch 3: 0.20000000298023224
Accuracy on batch 4: 0.20000000298023224
Accuracy on batch 5: 0.10000000149011612
Accuracy on batch 6: 0.20000000298023224
Accuracy on batch 7: 0.4000000059604645
Accuracy on batch 8: 0.20000000298023224
Accuracy on batch 9: 0.30000001192092896
Accuracy on all data: 0.1550000011920929, accelerator rank: 0
Accuracy on all data: 0.1550000011920929, accelerator rank: 1
```

File: README.md (hunk 2 above)
Comment (awaelchli) [reply README.md:182]: typo here: dpp -> ddp

Comment (awaelchli) [reply README.md:182]: and the if guard is necessary for mp.spawn

**Comment by Borda** [comment]
I see we shall have all examples tested in TM if possible
@SkafteNicki mind send a quick fix?
@awaelchli feel free to request changes not just commenting if you feel so... which would block auto-merge

**Comment by SkafteNicki** [comment]
Checkout fix in https://github.com/PyTorchLightning/metrics/pull/123

**Pull request merged**